homeassistant/components/tplink/const.py

```diff
@@ -1,5 +1,74 @@
 """Const for TP-Link."""
 import datetime
 
+from homeassistant.components.sensor import ATTR_STATE_CLASS, STATE_CLASS_MEASUREMENT
+from homeassistant.components.switch import ATTR_CURRENT_POWER_W, ATTR_TODAY_ENERGY_KWH
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_FRIENDLY_NAME,
+    ATTR_UNIT_OF_MEASUREMENT,
+    ATTR_VOLTAGE,
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+)
+
 DOMAIN = "tplink"
+COORDINATORS = "coordinators"
+
 MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(seconds=8)
+MAX_DISCOVERY_RETRIES = 4
+
+ATTR_CONFIG = "config"
+ATTR_TOTAL_ENERGY_KWH = "total_energy_kwh"
+ATTR_CURRENT_A = "current_a"
+
+CONF_MODEL = "model"
+CONF_SW_VERSION = "sw_ver"
+CONF_EMETER_PARAMS = "emeter_params"
+CONF_DIMMER = "dimmer"
+CONF_DISCOVERY = "discovery"
+CONF_LIGHT = "light"
+CONF_STRIP = "strip"
+CONF_SWITCH = "switch"
+CONF_SENSOR = "sensor"
+
+PLATFORMS = [CONF_LIGHT, CONF_SENSOR, CONF_SWITCH]
+
+ENERGY_SENSORS = {
+    ATTR_CURRENT_POWER_W: {
+        ATTR_UNIT_OF_MEASUREMENT: POWER_WATT,
```

Comment:
We got some cool new helper classes just for this! 

```python
from homeassistant.components.sensor import SensorEntityDescription
```

With that class (which you can extend if we don't cover everything you need), you can also just set it on the entity and the entity base class will figure out the static vars:

```python
class TPLinkSensor(SensorEntity):
    def __init__(self, description):
        self.entity_description = description
```

Reply:
nice ... will try it 👍